mongodbatlas/resource_project_api_key.go

```diff
@@ -79,42 +79,41 @@ type APIProjectAssignmentKeyInput struct {
 	RoleNames []string `json:"roles,omitempty"`
 }
 
+const errorNoProjectAssignmentDefined = "could not obtain a project id as no assignments are defined"
+
 func resourceMongoDBAtlasProjectAPIKeyCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	conn := meta.(*config.MongoDBClient).Atlas
-	projectID := d.Get("project_id").(string)
-	createRequest := new(matlas.APIKeyInput)
 
 	var apiKey *matlas.APIKey
 	var err error
 	var resp *matlas.Response
 
+	createRequest := new(matlas.APIKeyInput)
 	createRequest.Desc = d.Get("description").(string)
 	if projectAssignments, ok := d.GetOk("project_assignment"); ok {
 		projectAssignmentList := ExpandProjectAssignmentSet(projectAssignments.(*schema.Set))
-		for _, apiKeyList := range projectAssignmentList {
-			if apiKeyList.ProjectID == projectID {
-				createRequest.Roles = apiKeyList.RoleNames
-				apiKey, resp, err = conn.ProjectAPIKeys.Create(ctx, projectID, createRequest)
-				if err != nil {
-					if resp != nil && resp.StatusCode == http.StatusNotFound {
-						d.SetId("")
-						return nil
-					}
-				}
+
+		// creates api key using project id of first defined project assignment
+		firstAssignment := projectAssignmentList[0]
```

Comment:
do we need to check len(projectAssignmentList) > 0?

Reply:
The schema enforces that project_assignment must be defined with MinItem of 1 so there is no need really